### ISSUE TITLE
How is the classification performed?

### ISSUE THREAD

**Issue by aflah02** [ChrisIsKing/Rule-By-Example#2]
Hey!
I am curious about the classification setup since I can't find anything about it in the paper. I would assume for BERT baseline it's a MLP being trained but what about the supervised case of RBE? Are the representations from pretrained models used to train a classifier or is it just distance based? Can you like explain the testing time process for both supervised and unsupervised cases for a new sample. That would probably clear it up for me.

**Comment by ChrisIsKing** [comment]
For RBE It is a distance based classification based on the representations from the `Rule Encoder` and the `Text Encoder`. The testing time process is the same as training. First exemplars are extracted from the ruleset, if no rules fire a random sample of exemplars from the ruleset is taken. The exemplars are passed through the `Rule Encoder`, text through the `Text Encoder`. Classification can be performed using any distance based metric e.g Cosine Similarity, Dot Product
